File: cpp/34e_1352A_SumOfRoundNumbers.cpp
#include<bits/stdc++.h>

int main(){
    int t, j;
    std::string n;
    std::vector<std::string> numb;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        numb.push_back(n);
    }
    for(int i = 0; i < t; i++){
        std::vector<std::string> ans;
        j = numb[i].size() - 1;
        n = "";
        while(j >= 0){
            if(numb[i][j] != '0'){
                n += numb[i][j];
                for(int k = j; k< (numb[i].size() - 1); k++){
                    n += "0";
                }
                ans.push_back(n);
                n = "";
            }
            j--;
        }
        std::cout<<ans.size()<<"\n";
        for(auto i : ans){
            std::cout<<i<<" ";
        }
        std::cout<<"\n";
    }
    return 0;
}

File: python/34e_1352A_SumOfRoundNumbers.py
def main():
    t = int(input())
    numb = []
    for _ in range(t):
        numb.append(input().strip())
    for i in numb:
        j = len(i) - 1
        ans = []
        n = ""
        while j >= 0:
            if i[j] != "0":
                n += i[j]
                for k in range(j, len(i) - 1):
                    n += "0"
                ans.append(n)
                n = ""
            j -= 1
        print(len(ans))
        for k in ans:
            print(k, end = " ")
        print()

if __name__ == "__main__":
    main()